File: progressBar/progressBar.py
import math
import time

def printProgressBar(progress, update):
    progressInc = math.floor(progress / 5)
    progressBar = '#' * progressInc + ' ' * (20 - progressInc)
    spinnerState = update % 4
    spinner = ' '

    match spinnerState:
        case 0:
            spinner = '|'
        case 1:
            spinner = '/'
        case 2:
            spinner = '-'
        case 3:
            spinner = '\\'

    print(f'\rProgress: ({spinner}) [{progressBar}] {progress}%', end='')

progress = 0
update = 0

while progress <= 100:
    printProgressBar(progress, update)
    time.sleep(0.250)
    progress += 5
    update += 1

print('\nComplete')

File: progressBar/progressBar.js
'use strict';

function printProgressBar(progress, update) {
    const progressInc = Math.floor(progress / 5);
    const progressBar = ''.concat('#'.repeat(progressInc), ' '.repeat(20 - progressInc));
    const spinnerState = update % 4;
    let spinner = ' ';

    switch (spinnerState) {
        case 0: spinner = '|';
                break;
        case 1: spinner = '/';
                break;
        case 2: spinner = '-';
                break;
        case 3: spinner = '\\';
                break;
    }

    process.stdout.write(`\rProgress: (${spinner}) [${progressBar}] ${progress}%`);
}

async function updateProgressBar() {
    return new Promise(async (resolve) => {
        let progress = 0;
        let update = 0;

        while (progress <= 100) {
            printProgressBar(progress, update);
            await new Promise((resolve) => {
                setTimeout(resolve, 250);
            });
            progress += 5;
            update++;
        }

        resolve();
    });
}

(async function() {
    await updateProgressBar();
    process.stdout.write('\nComplete\n');
})();
